app/controllers/admin/loaned_books_controller.rb

```diff
@@ -73,4 +86,27 @@ def check_valid_user
     flash.now[:danger] = t "error.user_invalid"
     render :new
   end
+
+  def check_status_params
+    status_value = LoanedBook.statuses.values
+    return if status_value.include?(params[:status].to_i)
+
+    render json: {error: I18n.t("error.status_not_exist")}
+  end
+
+  def update_status_loaned_detail params
+    @loaned_book.loaned_details.each {|loan| loan.update(status: params)}
```

Comment:
```suggestion
    @loaned_book.loaned_details.each {|loan| loan.update!(status: params)}
```
dùng update! nếu em làm với transaction

chị nghĩ chỗ ni chỉ update với những thèn loaned detail ko phải ở trạng thái cancel thôi chứ nhỉ
Những thèn ở trạng thái cancel tức là mình request mượn xong mình hủy (chị mượn 2, nhưng sau đó chị chỉ lấy 1 cuốn), thì cái cancel ko cần thiết phải tính vào chứ

còn nếu em có logic là ko dc cancel trong details mà chỉ cancel bên ngoài (tức là quay xe, ko mượn nữa) thì field status ở loaned details là ko cần thiết